### PR TITLE
Update ui

### PR DESCRIPTION
**Pull request by holdemani**
I'll formally request a review after it's polished, but it works so I guess you can start taking peeks at it.

### CODE REVIEW

File: Astral Pathfinder/Game/PlanetManager.hpp [hunk 1]
```diff
@@ -41,8 +41,12 @@ class PlanetManager {
   // MARK: - PlanetManager Methods
   Planet getPlanet(int n);
   Planet getSelectedPlanet();
-  void setPlanetDepoPercent(int p);
-  void setPlanetFertPercent(int p);
+  Planet getCurrentPlanet();
```
Comment (LoganTHarvell) [inline Astral Pathfinder/Game/PlanetManager.hpp]: Rename getDockedPlanet() ?

File: Astral Pathfinder/Game/UIManager.hpp [hunk 2]
```diff
@@ -13,6 +13,15 @@
 #include "Game.hpp"
 #include "PlanetManager.hpp"
 #include "PlanetInfo.hpp"
+#include "ShipInfo.hpp"
+
+// MARK: - UIManager Parameters
+
+namespace UiParamters {
```
Comment (LoganTHarvell) [inline Astral Pathfinder/Game/UIManager.hpp]: Fix typo

File: Astral Pathfinder/Game/UIManager.cpp [hunk 3]
```diff
@@ -73,12 +69,104 @@ void UIManager::render(Game::State *gameState) {
...
+      
+      if(movement) {
+        pm->setPlanetInfraPercent(100-percent, window);
+        pm->setPlanetReservePercent(percent, window);
+      }
+    }
+  }
 }
 
+int UIManager::checkClickedArea(SDL_Point p) {
```
Comment (LoganTHarvell) [inline Astral Pathfinder/Game/UIManager.cpp]: Maybe use enum to clarify what is being returned?

File: Astral Pathfinder/Game/UIManager.cpp [hunk 4]
```diff
@@ -73,12 +69,104 @@ void UIManager::render(Game::State *gameState) {
...
+void UIManager::setCurrentPlanet(Planet p) {
+  currentPlanetInfo.setUiTextures(p);
+}
+
+void UIManager::handleMouseDown(Game::State *gs, PlanetManager *pm) {
+  // If mouse button not pressed down, don't check for slider movement
+  if(!gs->mouseDown) return;
+  
+  if(gs->mouseDown)
+    window = checkClickedArea(gs->clickLocation);
```
Comment (LoganTHarvell) [inline Astral Pathfinder/Game/UIManager.cpp]: I think the code could be changed to enhance readability in this section. It isn't clear what one and two means in regards to the sliders.

File: Astral Pathfinder/Game/UIManager.hpp [hunk 5]
```diff
@@ -21,18 +30,24 @@ class UIManager {
...
   
 private:
   // MARK: - UIManager Fields
   // TODO: Implement main menu and endscreen
   // MainMenu mainMenu;
-  PlanetInfo planetInfo;
+  PlanetInfo selectedPlanetInfo, currentPlanetInfo;
+  ShipInfo shipInfo;
+  int window = -1;
+  bool oneClean = true, twoClean = true;
```
Comment (LoganTHarvell) [inline Astral Pathfinder/Game/UIManager.hpp]: This is very unclear

File: Astral Pathfinder/UI/PlanetInfo.cpp [hunk 6]
```diff
@@ -55,66 +81,141 @@ void PlanetInfo::clean() {
 void PlanetInfo::setUiTextures(Planet p) {
   setBoxes(p);
   
-  if (slider.isInitialized()) {
-    // Sets correct slider position for given planet
-    float ratio = p.getFertilityPercent()/100.0f;
-    SDL_Rect tmpR = slider.getBaseRect();
-    slider.setSliderPosition(static_cast<int>(tmpR.w * ratio));
+  if (!sliderOne.isInitialized() && !sliderTwo.isInitialized() && p.getPopulation() != 0) {
```
Comment (LoganTHarvell) [inline Astral Pathfinder/UI/PlanetInfo.cpp:84]: Is it possible that one is initialized and one isn't? That would cause issues.

**Review by LoganTHarvell — DISMISSED** [review]
I think some readability changes would be beneficial, especially in regards to the random 1's and 2's used in the code for representing either current planet or selected planet.

File: Astral Pathfinder/UI/PlanetInfo.cpp (hunk 6 above)
Comment (holdemani) [reply Astral Pathfinder/UI/PlanetInfo.cpp:84]: Nah, it's not possible. But I believe this was code you wrote, so I just modified it.

File: Astral Pathfinder/Game/PlanetManager.hpp (hunk 1 above)
Comment (holdemani) [reply Astral Pathfinder/Game/PlanetManager.hpp]: Done

File: Astral Pathfinder/Game/UIManager.hpp (hunk 2 above)
Comment (holdemani) [reply Astral Pathfinder/Game/UIManager.hpp]: Done

File: Astral Pathfinder/Game/UIManager.cpp (hunk 3 above)
Comment (holdemani) [reply Astral Pathfinder/Game/UIManager.cpp]: Done

File: Astral Pathfinder/Game/UIManager.hpp (hunk 5 above)
Comment (holdemani) [reply Astral Pathfinder/Game/UIManager.hpp]: Changed

File: Astral Pathfinder/UI/PlanetInfo.cpp (hunk 6 above)
Comment (holdemani) [reply Astral Pathfinder/UI/PlanetInfo.cpp:84]: This needs a lil reworking

**Comment by holdemani** [comment]
Don't merge. There's a new memory leak somewhere

**Comment by holdemani** [comment]
We're all good in the hood now

File: Astral Pathfinder/Game/UIManager.hpp [hunk 7]
```diff
@@ -37,17 +37,24 @@ class UIManager {
...
-  PlanetInfo selectedPlanetInfo, currentPlanetInfo;
+  PlanetInfo selectedPlanetInfo, DockedPlanetInfo;
   ShipInfo shipInfo;
-  int window = -1;
-  bool oneClean = true, twoClean = true;
+  enum {
+    none, currentPlanetWindow, selectedPlanetWindow
+  } currentWindow = none;
+  
+  enum {
```
Comment (LoganTHarvell) [inline Astral Pathfinder/Game/UIManager.hpp:46]: This can be declared inside of the handleMouseDown() method since that's the only m it is used.

**Review by LoganTHarvell — APPROVED** [review]

**Pull request merged**